AK/Platform.h

```diff
@@ -145,10 +145,14 @@
 #ifdef NAKED
 #    undef NAKED
 #endif
-#if !ARCH(AARCH64)
+#if !ARCH(AARCH64) || defined(AK_COMPILER_CLANG)
 #    define NAKED __attribute__((naked))
 #else
-#    define NAKED
+// GCC doesn't support __attribute__((naked)) on AArch64. We use NAKED to mark functions
+// that are entirely written in inline assembly; having these be inlined would cause
+// various problems, such as explicit `ret` instructions accidentally exiting the caller
+// function or GCC discarding function arguments as they appear "dead".
+#    define NAKED NEVER_INLINE
 #endif
```

Comment:
i feel that itʼd be better to say something about GCC + Aarch64 than GCC + Clang in the if clause… just as comment does.